app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -86,8 +86,18 @@ EOF
 elif [ -f /etc/debian_version ]; then
   <%= save_to_file('/etc/apt/sources.list.d/foreman_registration.list', @repo) %>
 <% if @repo_gpg_key_url.present? -%>
-  apt-get -y install ca-certificates gpg
-  curl --silent --show-error --output /etc/apt/trusted.gpg.d/client.asc <%= shell_escape @repo_gpg_key_url %>
+<% # "apt 1.2.35" on Ubuntu 16.04 does not support storing GPG public keys in "/etc/apt/trusted.gpg.d/" in ASCII format -%>
+<%
+  use_gpg_dearmor = @host.operatingsystem.name == 'Ubuntu' && @host.operatingsystem.major.split('.')[0].to_i < 18
+-%>
+
+  $PKG_MANAGER_INSTALL ca-certificates curl gnupg
```

Comment:
You only need gnupg when `use_gpg_dearmor` is true. It pulls in quite a few dependencies so I'd avoid it if possible.

Reply:
Fixed.